pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobDetails.tsx

```diff
@@ -59,6 +68,260 @@ enum TabKeysEnum {
+    (tabId: string): void => {
+      searchParams.set("tab", tabId);
+      setCurrentTab(tabId);
+      history.replace({
+        ...history.location,
+        search: searchParams.toString(),
+      });
+    },
+    [history, searchParams],
+  );
+  useEffect(() => {
+    refreshUserSQLRoles();
+  }, [refreshUserSQLRoles]);
+  useEffect(() => {
+    if (job) {
+      setJobTerminal(isTerminalState(job.status));
+    }
+  }, [job]);
+
+  return (
+    <div className={jobCx("job-details")}>
+      <Helmet title={"Details | Job"} />
+      <div className={jobCx("section page--header")}>
+        <Button
+          onClick={() => history.goBack()}
+          type="unstyled-link"
+          size="small"
+          icon={<ArrowLeft fontSize={"10px"} />}
+          iconPosition="left"
+          className={commonStyles("small-margin")}
+        >
+          Jobs
+        </Button>
+        <h3 className={jobCx("page--header__title")}>{`Job ID: ${jobId}`}</h3>
+      </div>
+      <section className={jobCx("section section--container")}>
+        <Loading
+          loading={isLoading}
+          page={"job details"}
+          error={error}
+          render={() => (
+            <>
+              <section className={cardCx("summary-card")}>
+                <Row gutter={24}>
+                  <Col className="gutter-row" span={24}>
+                    <SqlBox
+                      value={job?.description ?? "Job not found."}
+                      size={SqlBoxSize.CUSTOM}
+                      format={true}
+                    />
+                  </Col>
+                </Row>
+              </section>
+              <Tabs
+                className={commonStyles("cockroach--tabs")}
+                defaultActiveKey={TabKeysEnum.OVERVIEW}
+                onChange={onTabChange}
+                activeKey={currentTab}
+                items={[
+                  {
+                    key: "overview",
+                    label: TabKeysEnum.OVERVIEW,
+                    children: <OverviewTabContent job={job} />,
+                  },
+                  !ccContext &&
+                    hasAdminRole && {
+                      key: "profiler",
+                      label: TabKeysEnum.PROFILER,
+                      children: (
+                        <JobProfilerViewV2
+                          jobID={jobId}
+                          onFetchExecutionDetailFiles={
+                            onFetchExecutionDetailFiles
+                          }
+                          onCollectExecutionDetails={onCollectExecutionDetails}
+                          onDownloadExecutionFile={onDownloadExecutionFile}
+                        />
+                      ),
+                    },
+                ]}
+              />
+            </>
+          )}
+        />
+      </section>
+    </div>
+  );
+}
+
+function OverviewTabContent({
+  job,
+}: {
+  job: JobResponse | null;
+}): React.ReactElement {
+  const messageColumns = [
```

Comment:
Seems like this can be declared outside of the component.

Reply:
It most certainly can, thanks!